### ISSUE TITLE
Cura 4.6.1 (and before) - Numbering of extruders confusing (0 and 1 or 1 and 2)

### ISSUE THREAD

**Issue by upatamby** [Ultimaker/Cura#7759]
**Application version**
Cura 4.6.1

**Platform**
Windows 10

**Printer**
Ultimaker 3 Extended

**Reproduction steps**
1. Make some settings, e.g. set Build Plate Adhesion Extruder to Extruder 2 and set Brim width to a non default value
2. Click on the profile field to change it and look at the warning window

**Screenshot(s)**
![image](https://user-images.githubusercontent.com/35468193/81982323-ee12e100-9631-11ea-94bd-3d33f365fc17.png)

**Actual results**
You see that the extruder numbering is weird: the Build Plate Adhesion Extruder is changed from 0 to 1 while the Brim Width applies to Extruder 2.

**Expected results**
Consistent numbering of the extruders: either 0 and 1 or 1 and 2, through the entire application.

**Comment by Ghostkeeper** [comment]
Indeed, it should be "1" and "2". The bug is that this dialogue displays the internal representation of the setting values, for instance the drop-down options (like Build Plate Adhesion Type) also show our internal representation rather than the normal one (which is in most cases the same except capitalised).

**Comment by upatamby** [comment]
There are more strange things: in the example below, Extruder 1 is referred to as Extruder 2
Isn't it time for a thorough clean up?
![image](https://user-images.githubusercontent.com/35468193/96379225-8f790280-118f-11eb-9926-3ef9d542eccc.png)

**Comment by GregValiant** [comment]
Is this still an issue in 5.12.1?

**Comment by upatamby** [comment]
Sorry, I missed the question "Is this still an issue in 5.12.1?". 
My answer: yes it is. The only 'improvement' is that there is no confusing extruder mentioned anymore. Now there is no extruder mentioned at all. 

<img width="670" height="511" alt="Image" src="https://github.com/user-attachments/assets/138a184d-70fb-46a7-84b1-adf09abd6c5a" />

**Comment by GregValiant** [comment]
When I change the Horizontal Expansions and the Build Plate Adhesion Extruder I get this:

<img width="638" height="617" alt="Image" src="https://github.com/user-attachments/assets/43f87e63-4053-4570-9d5a-2ef4a8208419" />

That looks right to me.

Please post a project file ("File | Save Project") that shows the problem you are having.

**Comment by upatamby** [comment]
Your image shows that Extruder 2 (displayed as "1") has been chosen as the extruder for the build plate adhesion. This is not very intuitive because the extruder numbering in Cura user interface is "1" and "2", not "0" and "1".

The image below shows that extruder 1 and 2 are displayed as 0 and 1 resp. The internal and displayed representation are mixed up.
<img width="1041" height="561" alt="Image" src="https://github.com/user-attachments/assets/4d04a86e-cd6a-4c5d-a91a-4934be63315a" />

But the real serious confusing starts here:
<img width="1047" height="628" alt="Image" src="https://github.com/user-attachments/assets/6bc57600-6424-4868-9741-8d379efe8b56" />

The property Alternate Extra Wall is true for Extruder 1 but false for Extruder 2.
Since Cura doesn't show the Extruder-number any more in this windows you have no idea for which extruder this setting has been changed. 
In the previous versions the extruder number was displayed between parenthesis, but the numbering was confusing while it was a mix up between the internal numbering and the numbering as it should be displayed. The initial problem is 'solved' in the easiest way possible by just omitting the reference to the extruder number. 
I would like the line:
   Alternate Extra Wall                        False     True
to read:
   Alternate Extra Wall (Extruder 1)   False     True
where "Extruder 1" refers to the extruder with the internal representation "0".
Similar, any change in a property for Extruder 2 should contain the reference "(Extruder 2)" that refers to the extruder with the internal representation 1.
I'm sorry if this is confusing, but I'm not the designer but just a user of this software.

**Comment by GregValiant** [comment]
The extruders are numbered as "Base 0" in the Gcode, but they are numbered as "Base 1" in the Cura user interface.  That is also true for PrusaSlicer, Orca and other slicers.
You may find it confusing, but it isn't a "bug".
Cura also numbers the layers in the gcode using "Base 0".  Other slicers don't bother to add layer numbers in the Gcode.

I'm going to go ahead and close this.  This is 6 years old, it's the only report, and although you may find it confusing, it isn't a bug.
